src/item_group.cpp

```diff
@@ -191,7 +191,12 @@ void Item_modifier::modify(item &new_item) const
             // spawn a "water (0)" item.
             new_item.charges = std::max( 1l, ch );
         } else if(t != NULL) {
-            new_item.charges = std::min(ch, t->max_charges);
+            const auto a = new_item.ammo_type();
+            const auto c = std::min(ch, t->max_charges);
```

Comment:
try and avoid such short variable names (i appreciate the existing code sets a bad example). you could consider naming this variable `qty`